### PR TITLE
Add scrolldir@1.2.2 w/ npm auto-update

### PR DESCRIPTION
**Pull request by mercyg**
Pull request for issue: #10657 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the URL and other related info like the popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/dollarshaveclub/scrolldir
 * Official website (optional, not the repository): https://dollarshaveclub.github.io/scrolldir/
 * NPM package url (optional): https://www.npmjs.com/package/scrolldir
 * License and its reference:  MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 17
   - Count of stars: 94
   - Count of forks: 1
 * NPM download stats (optional):
   - Downloads in the last day: 95
   - Downloads in the last week: 679
   - Downloads in the last month: 2,252

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by maruilian11** [comment]
@mercyg thank you for joining us. 
could you please also fill the lib info out in the init comment, like mention the issue number in `Pull request for issue: #10518`

File: ajax/libs/scrolldir/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "javascript",
+    "window"
+  ],
+  "author": "Patrick Fisher <patrick@pwfisher.com>",
+  "license": "MIT",
+  "homepage": "https://dollarshaveclub.github.io/scrolldir/",
+  "npmName": "scrolldir",
+  "npmFileMap": [
+    {
+      "basePath": "src",
```
Comment (maruilian11) [inline ajax/libs/scrolldir/package.json]: we only host pre-built files for browser, which should be the files under `dist`.

Comment (mercyg) [inline ajax/libs/scrolldir/package.json]: Hi @maruilian11,  do you mean to add auto-update config to a library?

Comment (pvnr0082t) [inline ajax/libs/scrolldir/package.json]: @mercyg If there are pre-built files under `dist` and ready for browser, those are the files we want. Let's change the basePath to `"dist"` here and use `"**/*` as the rule instead. Thanks.

Comment (pvnr0082t) [inline ajax/libs/scrolldir/package.json]: @mercyg If there are the pre-built files under `dist` folder and ready for browser, those are the files we want. Let's change the `basePath` to `"dist"` and use `"**/*` as the rule instead. Thanks.

Comment (mercyg) [inline ajax/libs/scrolldir/package.json]: @pvnr0082t I have changed the basePath, please review it

**Comment by mercyg** [comment]
@pvnr0082t I have squashed the commit, please review it

**Comment by PeterDaveHello** [comment]
@mercyg the indent looks wrong, I'll suggest you to read the error message yourself first, and see if you can fix it or you need further help, so the contribution and review process can be more efficiently :)

**Comment by PeterDaveHello** [comment]
btw @mercyg can you fill the library info in the first comment? It's not a good idea to leave it blank.

**Comment by mercyg** [comment]
@PeterDaveHello I have tried using the JSONLint to validate the package.json and it returns a valid JSON but I when I do the npm test it saying wrong indent. And there doesn't seem to be any instruction on how to use the tool in: tool/fixFormat.js. Could you please help me on how to do that?

**Comment by PeterDaveHello** [comment]
@mercyg  JSONLint doesn't check the indent at all, not sure if you have nodejs and npm installded, if so, just run `npm install` at the root of cdnjs, and runt the command `tool/fixFormat.js`, you should see the indent been fixed.

**Comment by mercyg** [comment]
@PeterDaveHello I have updated the indent

**Comment by maruilian11** [comment]
@mercyg the version in your commit msg is wrong, and please modify your first line of the commit msg in 50 charaters.
I suggest
```
Add scrolldir@1.2.2 w/ npm auto-update 

Close #10518, cc @pwfisher
```
please remember leave line 2 blank, thanks.

**Comment by mercyg** [comment]
@maruilian11 I changed the commit message please review it, thanks

**Comment by PeterDaveHello** [comment]
This PR is for issue #10518 (the comment)

**Pull request merged**